contracts/OpConsensusStateManager.sol

```diff
@@ -0,0 +1,127 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.0;
+
+import './IbcVerifier.sol';
+
+
+// OptimisticConsensusStateManager manages the optimistic consensus
+// states. It keeps track a configurable number of optimistic
+// consensus states. IBC module uses this class to access an
+// optimistic consensus state at a given height to perform membership
+// proof.
+library OptimisticConsensusStateManager {
+    enum OpConsensusStateStatus { NOT_APPLIED, PENDING, TRUSTED }
+
+    struct ManagedOpConsensusState {
+        OptimisticConsensusState state;
+        OpConsensusStateStatus status;
```

Comment:
Can prob get rid of status field completely. Time inserted is enough info on whether we can use a state for proofs.

Reply:
so far the only concern i have is about updating the fraud proof window. if updating fraud proof window is allowed, then an already trusted optimisitic consensus state may become pending again. need to think about it carefully to make sure that's not an issue.